File: dbinit.R
library(RSQLite)
db <- dbConnect(SQLite(), dbname="./docs/test.sqlite")
dbSendQuery(conn = db,"CREATE TABLE doctor (
	id integer PRIMARY KEY AUTOINCREMENT,
	login string,
	pw integer
);")
dbSendQuery(conn = db,"CREATE TABLE patient (
	id integer PRIMARY KEY AUTOINCREMENT,
	doctor_id integer,
	name string,
	sex binary,
	dateofbirth date
);")
dbSendQuery(conn = db,"CREATE TABLE survey (
	id integer PRIMARY KEY AUTOINCREMENT,
	questionary_id integer,
	patient_id integer,
	survey_date date,
	results string
);")
dbSendQuery(conn = db,"CREATE TABLE questionary (
	id integer PRIMARY KEY AUTOINCREMENT,
	date date
);")
dbSendQuery(conn = db,"CREATE TABLE question (
	id integer PRIMARY KEY AUTOINCREMENT,
	question string,
	type integer,
	answers string,
	critical boolean
);")
dbSendQuery(conn = db,"CREATE TABLE quest_quest (
	id integer PRIMARY KEY AUTOINCREMENT,
	questionary_id integer,
	question_id integer
);")
dbSendQuery(conn = db,"CREATE TABLE doc_quest (
	id integer PRIMARY KEY AUTOINCREMENT,
	questionary_id integer,
	doctor_id integer
);")
dbDisconnect(db)

File: dbinit.py


with open("./docs/Untitled_sqlite_create.sql", "r") as sql:
    cmd = ""
    with open("dbinit.R","w") as db:
        db.write('library(RSQLite)\n')
        db.write('db <- dbConnect(SQLite(), dbname="./docs/test.sqlite")\n')     

        for line in sql:
            if line == "\n":
                continue
            elif line == ");\n":
                cmd += line.strip()
                db.write('dbSendQuery(conn = db,"' + cmd + '")\n')
                cmd = ""
            else:
                cmd += line
        db.write('dbDisconnect(db)')

